### PR TITLE
componentize query page checkboxes

### LINKED ISSUE #124: Replace checkboxes on customize query page with Checkbox component
## Description:

We introduced a checkbox component as part of [this PR](https://github.com/CDCgov/dibbs-query-connector/pull/120/files). We should go back and use this on the customize query page where we're hand rolling the checkbox in the markup.

## Acceptance Criteria:

* The checkbox component code is swapped out with the custom code on the customize query page.
  * The component may need to be expanded (new props, new variation takes, etc.) depending on the use case on that page
* Existing functionality on that page still works

### PR DESCRIPTION
**Pull request by fzhao99**
# PULL REQUEST

## Summary
Refactors the customize query page to use the checkbox component introduced in another PR to fix some accessibility issues related to making the checkbox via CSS.

Also removed a demo test file that we don't need

Also added a config option to set the local FHIR server url for use for in local e2e runs

## Related Issue

Fixes #124 

## Additional Information
Because the checkbox on this page is different than the out-of-box checkbox we have from USWDS, made a vanity component wrapper for use only on this page. Can chat with design to see if we need this elsewhere.

### For design
This is changing some of the frontend code for checkbox stylings, so just double check that the checkboxes on the query customization screen [match Figma](https://www.figma.com/design/Iuw9me6kYftBF4WTCEsCZz/Query-Connector?node-id=17-8697&t=ghBdLLJ4N9G01RnI-4).

Screenshot 2024-12-02 at 10 37 58 AM

## Checklist

- [ ] Descriptive Pull Request title
- [ ] Link to relevant issues
- [ ] Provide necessary context for design reviewers
- [ ] Update documentation

### CODE REVIEW

File: query-connector/src/app/fhir-servers.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ type FHIR_SERVER_CONFIG = {
 /**
  * The configurations for the FHIR servers currently supported.
  */
+const localE2EFhirServer =
+  process.env.E2E_LOCAL_FHIR_SERVER ?? "http://hapi-fhir-server:8080/fhir";
```
Comment (fzhao99) [inline query-connector/src/app/fhir-servers.ts:17]: let me know if anyone forsees any issues with this

**Comment by mikang** [comment]
Styling looks good! Thanks @fzhao99 !

**Review by robertandremitchell — COMMENTED** [review]
works for me. are we getting rid of the minus state? 
Screenshot 2024-12-02 at 2 59 13 PM

**Comment by fzhao99** [comment]
> works for me. are we getting rid of the minus state? 
ahhh good catch. I'll go back and add this

**Comment by fzhao99** [comment]
@robertandremitchell 

ready for re-review after adding back in the minus state!

**Review by robertandremitchell — APPROVED** [review]

**Pull request merged**